usefulUtilities/voteApp/appResources/voteApp.js

```diff
@@ -0,0 +1,1022 @@
+        visited: false,
+        domains: [],
+        voted: {
+            avatar: "", domain: ""
+        }
+    };
+
+    // UI variables
+    var ui;
+    var dataStore = {
+        unload: false,
+        loading: true,
+        loggedin: true,
+
+        visited: false, // set by zone HOLIDAY APP
+
+        voted: {
+            domain: false,
+            avatar: false
+        },
+        openPolls: {
+            avatar: false,
+            domain: false
+        },
+        visitedAllDomains: false, // for domains
+        domains: [],
+        avatars: []
+    };
+    
+
+    var domains = {
+        DOMAIN: "domain",
+        domainsInfo: {},
+        sendVote: function (name) {
+
+            var _this = this;
+
+            if (DEBUG) {
+                print("Domains sendVote");
+            }
+
+            _this.domainsInfo[name.toLowerCase()].voted = true;
+            dataStore.voted.domain = true;
+
+            if (DEBUG) {
+                print("Domains sendVote voted name", JSON.stringify(dataStore.voted.domain));
+            }
+
+            updateUI();
+
+            var params = utils.encodeURLParams({
+                type: VOTE_GOOGLE,
+                time: Date.now(),
+                uuid: MyAvatar.sessionUUID,
+                username: AccountServices.username,
+                name: name,
+                eventName: EVENT_NAME,
+                eventDate: EVENT_DATE,
+                contestName: DOMAIN
+            });
+            var onComplete = function (request) {
+
+                var voteAppSettings = utils.getVoteAppSettings();
+                voteAppSettings.voted[DOMAIN] = name.toLowerCase();
+                Settings.setValue(VOTE_APP_SETTINGS_NAME, voteAppSettings);
+
+                if (DEBUG) {
+                    print("Domain sendVote onComplete", JSON.stringify(voteAppSettings));
+                }
+
+            };
+            var onError = function () {
+                if (DEBUG) {
+                    print("Error in VoteApp.js: Issue in sendDomainVote()");
+                }
+
+                _this.domainsInfo[name.toLowerCase()].voted = false;
+                dataStore.voted.domain = false;
+
+                updateUI();
+            };
+
+            utils.sendRequest(GOOGLE_SCRIPTS_URL, params, onComplete, onError);
+        },
+
+        setDomains: function (gDomains) {
+            // [ "TheSpot", "Studio" ],
+            // add domains to the location array
+
+            var _this = this;
+
+            if (DEBUG) {
+                print("Domains setDomains");
+            }
+
+            var changed = false;
+            var existingDomains = Object.keys(_this.domainsInfo);
+
+            if (DEBUG) {
+                print("Domains existingDomains", JSON.stringify(existingDomains));
+                print("Domains NEW google domain list", JSON.stringify(gDomains));
+            }
+
+            gDomains.forEach(function (domainName) {
```

Comment:
this inline function should be its own named function outside of this nested function

Reply:
perhaps something called `parseDomainData` or similar?